include/libdnf/rpm/repo.hpp

```diff
@@ -118,262 +109,349 @@ struct Repo {
-
-    /**
-    * @brief Try to preserve remote side timestamps
-    *
-    * When set to true the underlying librepo is asked to make an attempt to set the timestamps
-    * of the local downloaded files (repository metadata and packages) to match those from
-    * the remote files.
-    * This feature is by default switched off.
-    *
-    * @param preserveRemoteTime true - use remote file timestamp, false - use the current time
-    */
-    void setPreserveRemoteTime(bool preserveRemoteTime);
-    bool getPreserveRemoteTime() const;
-
-    const std::vector<std::string> & getContentTags();
-    const std::vector<std::pair<std::string, std::string>> & getDistroTags();
-
-    /**
-    * @brief Get list of relative locations of metadata files inside the repo
-    *
-    * e.g. [('primary', 'repodata/primary.xml.gz'), ('filelists', 'repodata/filelists.xml.gz')...]
-    *
-    * @return vector of (metadata_type, location) string pairs
-    */
-    const std::vector<std::pair<std::string, std::string>> getMetadataLocations() const;
-
-    std::string getCachedir() const;
-    void setRepoFilePath(const std::string & path);
-    const std::string & getRepoFilePath() const noexcept;
-    void setSyncStrategy(SyncStrategy strategy);
-    SyncStrategy getSyncStrategy() const noexcept;
-    void downloadUrl(const char * url, int fd);
-
-    /**
-    * @brief Set http headers.
-    *
-    * Example:
-    * {"User-Agent: Agent007", "MyMagicHeader: I'm here", nullptr}
-    *
-    * @param headers nullptr terminated array of C strings
-    */
-    void setHttpHeaders(const char * headers[]);
-
-    /**
-    * @brief Get array of added/changed/removed http headers.
-    *
-    * @return nullptr terminated array of C strings
-    */
-    const char * const * getHttpHeaders() const;
-    std::vector<std::string> getMirrors() const;
-
-    void setSubstitutions(const std::map<std::string, std::string> & substitutions);
+
+    /// @replaces libdnf:repo/Repo.hpp:method:Repo.setMaxMirrorTries(int maxMirrorTries)
+    void set_max_mirror_tries(int max_mirror_tries);
+
+    /// Gets timestamp of metadata "primary" file
+    /// @replaces libdnf:repo/Repo.hpp:method:Repo.getTimestamp()
+    int64_t get_timestamp() const;
+
+    /// Gets the highest timestamp from repomd records
+    /// @replaces libdnf:repo/Repo.hpp:method:Repo.getMaxTimestamp()
+    int get_max_timestamp();
+
+    /// Try to preserve remote side timestamps
+    /// When set to true the underlying librepo is asked to make an attempt to set the timestamps
+    /// of the local downloaded files (repository metadata and packages) to match those from
+    /// the remote files.
+    /// This feature is by default switched off.
+    /// @param preserveRemoteTime true - use remote file timestamp, false - use the current time
+    /// @replaces libdnf:repo/Repo.hpp:method:Repo.setPreserveRemoteTime(bool preserveRemoteTime)
+    void set_preserve_remote_time(bool preserve_remote_time);
```

Comment:
How does it work? I download metadata by reposync command to a dnf cache or at different place? Then when I try to refresh metadata, time stamp is overridden?